protocols/renegadex.js

```diff
@@ -0,0 +1,221 @@
+        },
+        bAutoBalanceTeams: { // alias of 'bSpawnCrates'
+          type: 'boolean',
+          description: 'Whether the server will spawn crates in this game for balancing'
+        },
+        bSpawnCrates: {
+          type: 'boolean',
+          description: 'Whether the server will spawn crates in this game for balancing'
+        },
+        CrateRespawnAfterPickup: {
+          type: 'integer',
+          minimum: 0,
+          description: 'interval for crate respawn (after pickup)'
+        }
+      },
+      required: [
+        'Player Limit',
+        'Time Limit',
+        'Team Mode',
+        'Game Type',
+        'Vehicle Limit',
+        'Mine Limit'
+      ]
+    }
+  }
+}
+export const MasterServerResponseSchema = {
+  type: 'array',
+  items: { $ref: '#/$defs/server' },
+  $defs: {
+    server: MasterServerServerInfoSchema
+  }
+}
+
+/**
+ * Implements the protocol for Renegade X, an UnrealEngine3 based game, using a custom master server
+ */
+export default class renegadex extends Core {
+  constructor () {
+    super()
+    this.usedTcp = true
+  }
+
+  async run (state) {
+    const servers = await this.request({
+      url: 'https://serverlist-rx.totemarts.services/servers.jsp',
+      responseType: 'json'
+    })
+
+    if (!servers) {
+      throw new Error('Unable to retrieve master server list')
+    }
+
+    // TODO: Ajv response validation
+    // const isDataValid = ajv.validate(MasterServerResponseSchema, servers)
+    // if (!isDataValid) {
+    //   throw new Error(`Received master server data is unknown/invalid: ${ajv.errorsText(ajv.errors)}`)
+    // }
+
+    const serverInfo = servers.find(
+      (server) =>
+        server.IP === this.options.address && server.Port === this.options.port
+    )
+
+    if (serverInfo == null) {
+      throw new Error('Server not found in master server list')
+    }
+
+    // set state properties based on received server info
+    this.populateProperties(state, serverInfo)
+  }
+
+  /**
+   * Translates raw properties into known properties
+   * @param {Object} state Parsed data
+   */
+  populateProperties (state, serverInfo) {
+    let emptyPrefix = ''
+    if (serverInfo.NamePrefix) emptyPrefix = serverInfo.NamePrefix + ' '
+    const servername = `${emptyPrefix}${serverInfo.Name}`
+    const numplayers = serverInfo.Players || 0
+    const numbots = serverInfo.Bots || 0
+    const variables = serverInfo.Variables || {}
+
+    state.name = servername
+    state.map = serverInfo['Current Map']
+    state.password = variables.bPassworded
+
+    state.numplayers = numplayers
+    state.maxplayers = variables['Player Limit'] || 0
+
+    // due to master server not providing bot/player list, and standard result has no bot count, add list with dummy values
+    state.players = Array.from(new Array(numplayers).keys(), (i) => ({ name: `Player #${i + 1}`, raw: {} }))
+    state.bots = Array.from(new Array(numbots).keys(), (i) => ({ name: `Bot #${i + 1}`, raw: {} }))
+
```

Comment:
In v4 we had players::setNum, which just filled the players field with empty values if the server wouldn't provide them, but it has been removed in https://github.com/gamedig/node-gamedig/pull/389 as you couldn't really determine whether the server provided the values but are as they are presented or filled by that method, please remove these.

Reply:
The information of how many bots are set/playing whould be left out in the standard result, as #389 was only adding the player count. But yeah okay, could be something to improve/extend.

Reply:
Remove the lines. But also passed the raw server info as output now,